core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationStatusServiceSpec.js

```diff
@@ -0,0 +1,197 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit test for the Translation status service.
+ */
+
+describe('Translation status service', function() {
+  beforeEach(module('oppia', function($provide) {
+    $provide.value('TranslationLanguageService', {
+      getActiveLanguageCode: function() {
+        return 'en';
+      },
+      availableTranslationLanguageCodes: ['en']
+    });
+    $provide.value('ExplorationStatesService', {
+      isInitialized: function() {
+        return true;
+      },
+      getStateNames: function() {
+        return ['First', 'Second', 'Third'];
+      },
+      getContentIdsToAudioTranslationsMemento: function(stateName) {
+        if (stateName === 'First') {
```

Comment:
Create a states dict and for each state add whatever required for this tests (I think interactionId and citat). Try avoiding if - else here and below.

Reply:
> Create a states dict and for each state add whatever required for this tests (I think interactionId and citat). Try avoiding if - else here and below.
Hey @DubeySandeep .
Even if we create a dict, we still have different operations for different stateNames here right? How are we supposed to still perform those operations differently for different stateNames and eliminate the if - else here?

Reply:
done